### ISSUE TITLE
pml monitoring fails to build 

### ISSUE THREAD

**Issue by naughtont3** [open-mpi/ompi#4402]
### What version of Open MPI are you using? (e.g., v1.10.3, v2.1.0, git branch name and hash, etc.)

 - OMPI master `05f98d6`
 - Linux x86-64, git clone, but should apply to all

### Please describe the system on which you are running

* Operating system/version:  Linux
* Computer hardware: x86-64
* Network type: 

-----------------------------

## Details of the problem

The PML Monitoring framework is not building.  It silently says that it can not compile, e.g., output from config.log

```
   ...<snip>...
configure:5914: --- MCA component pml:monitoring (m4 configuration macro)
configure:325400: checking for MCA component pml:monitoring compile mode
configure:325406: result: dso
configure:325917: checking if MCA component pml:monitoring can compile
configure:325919: result: no
   ...<snip>...
```

Test that **should** work but does not:

```
shell$ cd test/monitoring/
shell$ mpirun -np 2 -mca pml_monitoring_enable 2 -mca pml_monitoring_enable_output 2 ./monitoring_test
```

**Comment by naughtont3** [comment]
@bosilca I think you are the proper person to take a look.  Thanks.

**Comment by bosilca** [comment]
Your patch in #4403 fixes the issue. Please fill a PR to cherry-pick it on the 3.1.x.

**Comment by naughtont3** [comment]
Ok, so patch in #4411 should fix this for 3.1.x
